datafusion/sql/src/parser.rs

```diff
@@ -43,6 +43,30 @@ fn parse_file_type(s: &str) -> Result<String, ParserError> {
 }
 
 /// DataFusion extension DDL for `CREATE EXTERNAL TABLE`
+///
+/// Syntax:
+///
+/// ```text
+/// CREATE EXTERNAL TABLE
+/// [ IF NOT EXISTS ]
+/// <TABLE_NAME>[ (<column_definition>) ]
+/// STORED AS <file_type>
+/// [ WITH HEADER ROW ]
+/// [ DELIMITER <char> ]
+/// [ COMPRESSION TYPE <GZIP | BZIP2 | XZ | ZSTD> ]
+/// [ PARTITIONED BY (<column list>) ]
+/// [ WITH ORDER (<ordered column list>)
+/// [ OPTIONS (<key_value_list>)
```

Comment:
The right bracket `]` seems missing.

Reply:
Good catch -- thank you !

```suggestion
/// [ WITH ORDER (<ordered column list>)
/// [ OPTIONS (<key_value_list>) ]
```